### DISCUSSION TITLE
Schema -> OOP Class

### DISCUSSION THREAD

**Discussion by charafsalmi** [schemaorg/schemaorg#3092] [Ideas]
Hi,

Thanks for the huge work you do there.

There should be a tool to automatically generate a class from any schema available on the website. This tool would allow to choose in which languages you want it, java, php, cpp, javascript, etc.

That would be a useful idea.

Best regards.

**Comment by thadguidry** [comment]
You first! 😉Sounds easy enough that you could write one, taking our Schema.org rdf file and make your client widget or IDE extension in Eclipse, VSCode, or JetBrains IDEs, or just write a Javascript webpage in whatever framework you enjoy that would parse our current rdf file, cache it locally in your browser storage, and run the interface locally in your web browser.  You can do it, we have confidence in you!

**Reply by charafsalmi** [reply]
Actually, I wouldn’t have expected a mocking reply from somebody as involved in respectable projects as you.

**Reply by thadguidry** [reply]
I sincerely encourage others to contribute when they can.  My response to you was pure encouragement and ideas for others to contribute and help you.

**Comment by danbri** [comment]
This direction has been explored. It has its uses but real world
applicability is high limited:

1.) it gets tightly bound to one version of its source data

2) it doesn’t deal with the fact that incoming edges to a node can be just
as useful and important as their inverses

Person3 alumniOf Org2
Vs
Org2 alumni Person3

…these are two ways of describing exactly the same state of affairs.

3) this approach generally makes it hard to extend descriptions using
external or private vocabulary, unless something like an rdf api is used
